src/rules/no-duplicates.js

```diff
@@ -2,21 +2,198 @@ import resolve from 'eslint-module-utils/resolve'
 import docsUrl from '../docsUrl'
 
 function checkImports(imported, context) {
-  for (let [module, nodes] of imported.entries()) {
-    if (nodes.size > 1) {
-      for (let node of nodes) {
-        context.report(node, `'${module}' imported multiple times.`)
+  for (const [module, nodes] of imported.entries()) {
+    if (nodes.length > 1) {
+      const message = `'${module}' imported multiple times.`
+      const [first, ...rest] = nodes
+      const sourceCode = context.getSourceCode()
+      const fix = getFix(first, rest, sourceCode)
+
+      context.report({
+        node: first.source,
+        message,
+        fix, // Attach the autofix (if any) to the first import.
+      })
+
+      for (const node of rest) {
+        context.report({
+          node: node.source,
+          message,
+        })
+      }
+    }
+  }
+}
+
+function getFix(first, rest, sourceCode) {
+  // Sorry ESLint <= 3 users, no autofix for you.
+  if (typeof sourceCode.getCommentsBefore !== 'function') {
+    return undefined
+  }
+
+  const defaultImportNames = new Set(
+    [first, ...rest].map(getDefaultImportName).filter(Boolean)
+  )
+
+  // Bail if there are multiple different default import names – it's up to the
```

Comment:
is there a way we could reduce the ones that are the same? ie, if the list was `a, a, b, c`, it could be autofixed to `a, b, c`?

Reply:
I don't think it's possible to end up with `a, a, b, c`, because duplicate identifiers are a syntax error: https://astexplorer.net/#/gist/bcba02f3c293bd99d3e00b3061befcfb/89b6673f21acde39324bb7063394b56834ede08e

Reply:
(Also, note that the particular line you commented on isn't about the specifiers inside `{...}`, but about _default_ imports: `import a from "a"; import b from "a"` cannot be autofixed.)